### PR TITLE
feat(go): support Go 1.20 and Delve 1.20.1

### LINKED ISSUE #125: Add `skaffold debug` support for go1.20
**Description of the issue**:

_This is a follow-up from the [Slack 🧵](https://kubernetes.slack.com/archives/CABQMSZA6/p1678829174059479)_ 

When using `skaffold debug` with a go1.20 compiled binary we see

```console
WARNING: undefined behavior - version of Delve is too old for Go version 1.20.1 (maximum supported version 1.19)
```

Attaching a debugger fails.

**Expected behavior**:

Attaching a debugging succeeds 🎉 

**Steps to reproduce**: 

In https://github.com/GoogleContainerTools/skaffold,

```bash
➜ minikube start
...
➜ cd integration/testdata/debug/go
➜ # update go version, and disable modules
➜ sed -i 's/golang:1.15/golang:1.20/g' Dockerfile
➜ sed -i 's/RUN go/RUN GO111MODULE=off go/g' Dockerfile
➜ cd ..
➜ skaffold debug
...
[go-web] WARNING: undefined behavior - version of Delve is too old for Go version 1.20.2 (maximum supported version 1.19)
```

**Environment**: 

**`skaffold` Configuration:**
```console
➜ skaffold version  
v2.2.0
```

**Log output**: 

When attempting to attach a debugger, we see
```console
dumpStacktrace - Failed to get debugger state Internal debugger error: <nil> not an Int
runtime.gopanic (0x435591)
	/usr/local/go/src/runtime/panic.go:884
go/constant.Int64Val (0x591873)
	/usr/local/go/src/go/constant/value.go:503
github.com/go-delve/delve/pkg/proc.(*Variable).parseG.func2 (0x74b688)
	/go/delve-source/pkg/proc/variables.go:885
github.com/go-delve/delve/pkg/proc.(*Variable).parseG (0x74b11b)
	/go/delve-source/pkg/proc/variables.go:921
github.com/go-delve/delve/pkg/proc.GetG (0x7476ad)
	/go/delve-source/pkg/proc/variables.go:259
github.com/go-delve/delve/service/api.ConvertThread (0x75d3f3)
	/go/delve-source/service/api/conversions.go:109
github.com/go-delve/delve/service/debugger.(*Debugger).state (0x7e8a64)
	/go/delve-source/service/debugger/debugger.go:581
github.com/go-delve/delve/service/debugger.(*Debugger).State (0x7e8438)
	/go/delve-source/service/debugger/debugger.go:550
github.com/go-delve/delve/service/rpc2.(*RPCServer).State (0x8639cb)
	/go/delve-source/service/rpc2/server.go:114
reflect.Value.call (0x4c7ce4)
	/usr/local/go/src/reflect/value.go:584
reflect.Value.Call (0x4c71db)
	/usr/local/go/src/reflect/value.go:368
github.com/go-delve/delve/service/rpccommon.(*ServerImpl).serveJSONCodec.func3 (0x942f99)
	/go/delve-source/service/rpccommon/server.go:373
runtime.goexit (0x4665a0)
	/usr/local/go/src/runtime/asm_amd64.s:1594
```

**Additional Information**: 

Since `skaffold-debug-support` uses [delve 1.9.1](https://github.com/GoogleContainerTools/container-debug-support/blob/duct-tape/go/helper-image/Dockerfile#L7), I think this could be resolved with an update to [delve 1.20](https://github.com/go-delve/delve/blob/master/CHANGELOG.md). It looks like there are similar updates [for delve 1.8](https://github.com/GoogleContainerTools/container-debug-support/pull/109) and [for go1.19](https://github.com/GoogleContainerTools/container-debug-support/pull/119).

### PR DESCRIPTION
**Pull request by sethrylan**
## What are you trying to accomplish?

Resolves https://github.com/GoogleContainerTools/container-debug-support/issues/125

## What approach did you choose?

I followed the pattern in https://github.com/GoogleContainerTools/container-debug-support/pull/119.

- Updated the helper image
- Updated the delve version
- Added tests for go1.20
- Ran `bash ../run-its.sh` ✅ 

## Anything you want to call attention to for reviewer?

- I did not update the go version in the [nodejs helper-image](https://github.com/GoogleContainerTools/container-debug-support/blob/7c11fe8803468d720811203485ea6477f3b0c832/nodejs/helper-image/Dockerfile#L2).

### CODE REVIEW

File: go/skaffold.yaml [hunk 1]
```diff
@@ -94,6 +94,14 @@ profiles:
...
             buildArgs:
               GOVERSION: 1.19
+      - op: add
+        path: /build/artifacts/-
+        value:
+          image: go120app
+          context: test/goapp
+          docker:
+            buildArgs:
+              GOVERSION: '1.20'
```
Comment (sethrylan) [inline go/skaffold.yaml:104]: The quotes are necessary. Without them the trailing zero is lost and the buildarg becomes `1.2`.

I'm not sure why that happens.

Comment (sethrylan) [inline go/skaffold.yaml:104]: The quotes are necessary. Without them the trailing zero is lost and the buildarg becomes `1.2`.

**Comment by aaron-prindle** [comment]
Thanks for the PR here @sethrylan!  I am still getting to up to speed on the CI/CD for this repo, seems there is a failing test on this PR currently but I don't think it is related to your PR changes (seem it is python env related).  Going to meet w/ the current `skaffold debug` maintainer @briandealwis and get more info so I can support this repo better and get this PR in.  Will update you when I know more, thanks

**Comment by sethrylan** [comment]
The error says

```console
 > [python27 3/6] RUN PYTHONUSERBASE=/dbgpy/pydevd/python2.7 pip install --user pydevd --no-warn-script-location:
#39 3.207       File "/tmp/pip-install-vQuf6F/pydevd/setup.py", line 53, in <module>
#39 3.207         import pydevd
#39 3.207       File "pydevd.py", line 3244
#39 3.207         def log_to(log_file:str, log_level=3) -> None:
#39 3.207                            ^
#39 3.207     SyntaxError: invalid syntax
```

I think this CI failure is related to pydevd's [recently dropped](https://github.com/fabioz/PyDev.Debugger/commit/4456092895a70eaf0c647393f1f776be71978f49) support for python versions <3.6.

@aaron-prindle, I've created [a PR](https://github.com/GoogleContainerTools/container-debug-support/pull/127) to fix this issue to get CI back to 🟢. Let me know what you think!

**Comment by aaron-prindle** [comment]
@sethrylan thanks for the CI fix 🙏, just merged your PR there (#127).  Can you rebase this PR now and see if we can get this green and merge it?

**Comment by sethrylan** [comment]
Updated and CI looks good! Let me know if there's anything else we need to do.

**Review by aaron-prindle — APPROVED** [review]
LGTM!  I think your other PR also fixed issues I was seeing in the PR to add Python 3.11 support (#122).  I'm going to get that in and then I will cut a release which will create the images and `skaffold debug` will automatically start consuming them

**Pull request merged**